core/src/main/cpp/util/Base64Secure.cpp

```diff
@@ -121,8 +121,17 @@ std::shared_ptr<ByteArray> Base64Secure::decode(const std::string& s) {
     StringConstIterator p = s.begin();
     while (p != s.end()) {
     	const char c = *p++;
+
+    	// Ensure the character is not "negative".
+    	if (c & 0x80) {
+    		invalid = true;
+    		continue;
+    	}
+
+    	// Lookup the character in the decoder map.
```

Comment:
All of the invalid = true cases skip in order to not exit early and leak information about a bad decode (since decoding is often related to crypto). There's no lookup in the map for negative characters or characters > 127. It would be a segfault if not for the bitmask `& 0x7F` but that changes the value if the high bit is set.